src/screens/Messages/Conversation/index.tsx

```diff
@@ -109,7 +109,10 @@ let Header = ({
     if (isWeb) {
       navigation.replace('Messages')
     } else {
-      navigation.pop()
+      navigation.reset({
+        index: 0,
+        routes: [{name: 'Messages'}],
+      })
```

Comment:
For now, in cases where we got to this screen from a notification, adding the `Messages` route to the start of this creates the "pop" animation that we are looking for.